certloader/spiffe_tls_config.go

```diff
@@ -111,8 +111,8 @@ func (c *spiffeTLSConfig) GetClientConfig() *tls.Config {
 func (c *spiffeTLSConfig) GetServerConfig() *tls.Config {
 	config := c.base.Clone()
 
-	// Only set client requirement if not disabled in base.
-	if config.ClientAuth == tls.RequireAndVerifyClientCert {
+	if !c.clientDisableAuth {
```

Comment:
The regular code-path was setting `config.ClientAuth` to `tls.RequireAndVerifyClientCert` and this was relying on it. The test is taking another path and therefore was set to `NoClientCert` instead, making the test fail.

This adjustment is getting rid of assumed code paths and will work on both.